### ISSUE TITLE
inconsistent message counts

### ISSUE THREAD

**Issue by xunam** [pazz/alot#463]
When opening a search buffer, the status bar indicates the number of messages matching the query and the buffer displays the summaries of threads containing such messages. When retagging messages, this message count is not updated consistently. The typical case is when removing the 'inbox' tag from a thread in the initial view: the message count is decremented by the total number of messages in the thread, not the number of messages that matched the query.

**Comment by varac** [comment]
I have a similar problem. One particular search doesn't show any messages (empty buffer), but the message count shows incorrectly 58 messages.
searching with notmuch for the same search terms shows 0 messages, but usign `--exclude=false` shows 58 messages as well.

So it seems the message count wrongly include messages tagged with one of the `exclude_tags`, while the search result doesn't (correctly).

**Comment by pazz** [comment]
this sounds like a bug. A quick inspection of the search buffer code suggests
that excluded tags should also be excluded when counting results for the
statusline.

**Comment by varac** [comment]
Is there an example notmuch database you know of that I could use to reproduce issues and report bugs without the need of anonymising ? That would really help debugging this issue.

**Comment by pazz** [comment]
yes it would. We don't have this yet. But there is a test corpus in the notmuch test suite..
